### PR TITLE
Test gap: dependency resolution edge cases (#1496)

### LINKED ISSUE #1496: Test gap: dependency resolution edge cases untested (cycles, missing deps, unblocking)
## Problem

`packages/database/src/task-store.ts` dependency functions lack edge case tests:

### areDependenciesMet() (lines 354-367)
No tests for:
- Circular dependency (A→B→A) — confirmed bug #1484
- Missing dependency ID (dep points to nonexistent task)
- Self-dependency (A depends on A)

### getUnblockedTasks() (lines 331-346)
No tests at all. Missing:
- Tasks with all deps complete → should appear
- Tasks with one incomplete dep → should not appear
- Tasks with circular deps → behavior undefined (should be handled)

### checkAndUnblock()
No tests for the cascade: completing a task should unblock its dependents.

### safeParseJsonArray()
No error path tests for corrupted JSON in dependsOn field.

## Suggested Tests

```ts
it("areDependenciesMet returns false for self-dependency", () => {
  createTask("a", "ws", "A", "", ["a"], ...);
  expect(areDependenciesMet("a")).toBe(false);
});

it("areDependenciesMet returns false for missing dependency", () => {
  createTask("a", "ws", "A", "", ["nonexistent"], ...);
  expect(areDependenciesMet("a")).toBe(false);
});

it("getUnblockedTasks excludes tasks with incomplete deps", () => {
  createTask("a", "ws", "A", "", [], ...);
  createTask("b", "ws", "B", "", ["a"], ...);
  expect(getUnblockedTasks("ws").map(t => t.id)).toEqual(["a"]);
});
```

## Priority

MEDIUM — confirmed bug #1484 would have been caught by these tests.

### PR DESCRIPTION
**Pull request by nick-pape**
## Summary

- Adds `json-helpers.test.ts` with full branch coverage for `safeParseJsonArray` (falsy/empty input, malformed JSON, valid non-array JSON, string arrays, mixed-type filtering) — this helper had **zero tests**
- Extends `areDependenciesMet` tests with self-dependency and mutual circular-dep (A→B→A) edge cases
- Adds `getUnblockedTasks` test: cyclic tasks never surface (inert, not crashing)
- Adds new `checkAndUnblock` describe block: alias-contract test and all-blocked fixture

**Note:** The bulk of #1496's ask (areDependenciesMet, getUnblockedTasks, detectDependencyCycle happy/sad paths) already shipped alongside the #1484 fix. This PR closes the residual gaps only. No production code changes.

## Test plan

- [x] `cd packages/database && npx vitest run` — all 351 tests pass (22 files, 21 new tests)
- [x] `rush build -t @grackle-ai/database` — build clean, no warnings

Closes #1496

### CODE REVIEW

File: common/changes/@grackle-ai/cli/nick-pape-1496-dependency-resolution-tests_2026-06-10-21-47.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "Add regression tests for dependency resolution edge cases (safeParseJsonArray, self-dep, circular deps, checkAndUnblock)",
+      "type": "patch",
```
[the remark this answers is not in the document]
Comment (nick-pape) [reply common/changes/@grackle-ai/cli/nick-pape-1496-dependency-resolution-tests_2026-06-10-21-47.json:5]: Dismissing — this repo's policy explicitly prohibits `"none"` in change files. All publishable packages use lockstep versioning and ship together, so the `@grackle-ai/cli` change file represents the entire lockstep group. Even test-only changes must use `patch` (or `minor`) to keep the version log accurate. Using `"none"` would suppress the version record entirely, which is not allowed here regardless of whether production code changed.
